homeassistant/components/waze_travel_time/config_flow.py

```diff
@@ -0,0 +1,143 @@
+                    ): cv.string,
+                    vol.Optional(
+                        CONF_EXCL_FILTER,
+                        default=self.config_entry.options.get(CONF_EXCL_FILTER),
+                    ): cv.string,
+                    vol.Optional(
+                        CONF_REALTIME,
+                        default=self.config_entry.options.get(
+                            CONF_REALTIME, DEFAULT_REALTIME
+                        ),
+                    ): cv.boolean,
+                    vol.Optional(
+                        CONF_VEHICLE_TYPE,
+                        default=self.config_entry.options.get(
+                            CONF_VEHICLE_TYPE, DEFAULT_VEHICLE_TYPE
+                        ),
+                    ): vol.In(VEHICLE_TYPES),
+                    vol.Optional(
+                        CONF_UNITS,
+                        default=self.config_entry.options.get(
+                            CONF_UNITS, self.hass.config.units.name
+                        ),
+                    ): vol.In(UNITS),
+                    vol.Optional(
+                        CONF_AVOID_TOLL_ROADS,
+                        default=self.config_entry.options.get(
+                            CONF_AVOID_TOLL_ROADS, DEFAULT_AVOID_TOLL_ROADS
+                        ),
+                    ): cv.boolean,
+                    vol.Optional(
+                        CONF_AVOID_SUBSCRIPTION_ROADS,
+                        default=self.config_entry.options.get(
+                            CONF_AVOID_SUBSCRIPTION_ROADS,
+                            DEFAULT_AVOID_SUBSCRIPTION_ROADS,
+                        ),
+                    ): cv.boolean,
+                    vol.Optional(
+                        CONF_AVOID_FERRIES,
+                        default=self.config_entry.options.get(
+                            CONF_AVOID_FERRIES, DEFAULT_AVOID_FERRIES
+                        ),
+                    ): cv.boolean,
+                }
+            ),
+        )
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Waze Travel Time."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: config_entries.ConfigEntry) -> OptionsFlow:
+        """Get the options flow for this handler."""
+        return OptionsFlow(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is not None:
+            await self.async_set_unique_id(
+                slugify(
+                    f"{DOMAIN}_{user_input[CONF_ORIGIN]}_{user_input[CONF_DESTINATION]}"
+                )
+            )
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(
+                title=self.config_entry.options.get(
+                    CONF_NAME,
```

Comment:
Just use a default prefix and origin and destination as title.

Reply:
I have updated this so we check `config_entry.data` and fall back to the default prefix to handle the scenario where it's defined in the import config